### PR TITLE
deprecate \Cake\Form\Form\errors();

### PR DESCRIPTION
**Pull request by inoas**
Ref https://github.com/cakephp/cakephp/issues/12596

### CODE REVIEW

**Comment by dereuromark** [comment]
Afaik the consensus was to focus on the actual functional splitting, and keeping the aliases of existing ones simple, so errors() instead of getErrors() - for the sake of changing it only this would introduce hundreds of more deprecations in the code, wouldnt it?
I am not opposed to it, especially when tools like rector could help to safely migrate.
But also not sure if we have to do that all for 4.0.

**Comment by inoas** [comment]
Yes, I wondered why there was not some shared interface.

File: src/Form/Form.php [hunk 1]
```diff
@@ -244,6 +244,24 @@ public function validate(array $data)
      * @return array Last set validation errors.
```
[the remark this answers is not in the document]
Comment (inoas) [reply src/Form/Form.php]: 3.7.0?

**Review by dereuromark — APPROVED** [review]

**Comment by inoas** [comment]
I got problems with CI.
Which branch do I need to base my PR on and against which branch should I PR?

**Comment by ADmad** [comment]
> Which branch do I need to base my PR on and against which branch should I PR?

You need to add your changes on top of `3.next` and make PR against `3.next` (which is already done).

File: tests/TestCase/ORM/EntityTest.php [hunk 2]
```diff
@@ -1183,30 +1183,30 @@ public function testErrors()
     {
         $this->deprecated(function () {
             $entity = new Entity();
-            $this->assertEmpty($entity->errors());
-            $this->assertSame($entity, $entity->errors('foo', 'bar'));
-            $this->assertEquals(['bar'], $entity->errors('foo'));
+            $this->assertEmpty($entity->getErrors());
+            $this->assertSame($entity, $entity->getErrors('foo', 'bar'));
```
Comment (ADmad) [inline tests/TestCase/ORM/EntityTest.php]: This should be a setter call `$entity->setErrors('foo', 'bar')`, that's what's causing tests to fail.

File: tests/TestCase/Form/FormTest.php [hunk 3]
```diff
@@ -150,6 +150,18 @@ public function testValidateDeprected()
      * @return void
      */
     public function testErrors()
+    {
+        $this->deprecated(function () {
+            $this->testGetErrors();
```
Comment (saeideng) [inline tests/TestCase/Form/FormTest.php]: you no need to wrap this method with `deprecated` when it does not use/have deprecated methods

**Review by saeideng — CHANGES_REQUESTED** [review]

**Comment by inoas** [comment]
Looks like the appveyor is unrelated.

**Comment by ADmad** [comment]
The `3.next` branch of manual also needs to be updated.

File: tests/TestCase/Form/FormTest.php [hunk 4]
```diff
@@ -140,16 +140,27 @@ public function testValidateDeprected()
...
 
     /**
      * Test the errors methods.
      *
      * @return void
+     * @deprecated 3.7.0 Use Form::testGetErrors() instead.
      */
     public function testErrors()
+    {
+        $this->testGetErrors();
```
Comment (garas) [inline tests/TestCase/Form/FormTest.php]: Why deprecate test method?

`testErrors()` should be removed now as there is new `testGetErrors()`

Now `testGetErrors()` runs twice

Comment (ADmad) [reply tests/TestCase/Form/FormTest.php]: Would you mind fixing it please :)

**Pull request merged**